From: SeongJae Park <sj@kernel.org>
Date: Mon, 16 Dec 2024 20:12:15 -0800
Subject: [PATCH] mm/damon: introduce DAMOS_STAT_FULL

DAMOS_STAT is a special DAMOS action that does nothing but only stat
accounting.  It skips applying DAMON operations layer-handled DAMOS
filters, since it could induce some overheads, while there was no stat
to account with the filtering results.  Now there are per-scheme and
per-region filter-passed bytes stats.  Since DAMOS_STAT's behavior is
documented and some users may still need it, introduce a new monitoring
purpose action, namely DAMOS_STAT_FULL.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index b88f710297a9..af28fa7c50a2 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -108,6 +108,7 @@ struct damon_target {
  * @DAMOS_MIGRATE_HOT:  Migrate the regions prioritizing warmer regions.
  * @DAMOS_MIGRATE_COLD:	Migrate the regions prioritizing colder regions.
  * @DAMOS_STAT:		Do nothing but count the stat.
+ * @DAMOS_STAT_FULL:	Do nothing but filtering and stat counting.
  * @NR_DAMOS_ACTIONS:	Total number of DAMOS actions
  *
  * The support of each action is up to running &struct damon_operations.
@@ -127,6 +128,7 @@ enum damos_action {
 	DAMOS_MIGRATE_HOT,
 	DAMOS_MIGRATE_COLD,
 	DAMOS_STAT,		/* Do nothing but only record the stat */
+	DAMOS_STAT_FULL,	/* Handle and account filters */
 	NR_DAMOS_ACTIONS,
 };
 
-- 
2.39.5

